packages/docs-theme/src/components/documentation.tsx

```diff
@@ -122,6 +122,18 @@ export interface DocumentationProps extends Props {
     tagRenderers: TagRendererMap;
 }
 
+// HACKHACK: this is brittle
+// detect Components page and subheadings
+const COMPONENTS_PATTERN = /\/components(\.[\w-]+)?$/;
+const CONTEXT_PATTERN = /\/context(\.[\w-]+)?$/;
+const HOOKS_PATTERN = /\/hooks(\.[\w-]+)?$/;
+const LEGACY_PATTERN = /\/legacy(\.[\w-]+)?$/;
+export const isNavSection = ({ route }: HeadingNode) =>
+    COMPONENTS_PATTERN.test(route) ||
+    CONTEXT_PATTERN.test(route) ||
+    HOOKS_PATTERN.test(route) ||
+    LEGACY_PATTERN.test(route);
```

Comment:
Moving this from `docs-app` to `docs-theme` feels a bit strange. I think `docs-theme` is meant to be generalizable outside of the Blueprint repo, and these regex patterns look specific to Blueprint's `docs-app`.

Reply:
ah I didn't understand what the distinction was for, I suppose some other project could rely on docs-theme to use the blueprint docs styling and provide their own content?

Reply:
Yup! I think that's the intention. I'm not sure how many other projects actually do that though.